prov/util/src/util_shm.c

```diff
@@ -81,8 +81,13 @@ void smr_cma_check(struct smr_region *smr, struct smr_region *peer_smr)
 	remote_iov.iov_len = sizeof(peer_smr->cma_cap);
 	ret = ofi_process_vm_writev(peer_smr->pid, &local_iov, 1,
 				    &remote_iov, 1, 0);
-	smr->cma_cap = (ret == -1) ? SMR_CMA_CAP_OFF : SMR_CMA_CAP_ON;
-	peer_smr->cma_cap = smr->cma_cap;
+
+	if (smr == peer_smr) {
+		smr->cma_cap_self = (ret == -1) ? SMR_CMA_CAP_OFF : SMR_CMA_CAP_ON;
```

Comment:
Isn't this initialization going to be skipped if a peer updates `smr->cma_cap` first by the check on L73?

Reply:
Yep, that's what aingerson@ found as well. Fixed in new revision.

Reply:
Oh, sorry, will check out new revision.